sherpa/utils/src/_utils.cc

```diff
@@ -84,13 +84,13 @@ static PyObject* ftest( PyObject* self, PyObject* args )
   if ( EXIT_SUCCESS != result.create( dof_1.get_ndim(), dof_1.get_dims() ) )
     return NULL;
 
-  double f, x;
-
   for ( npy_intp ii = 0; ii < nelem; ii++ ) {
 
-    f = ( chisq_1[ii] / dof_1[ii] ) / ( chisq_2[ii] / dof_2[ii] );
-    x = dof_2[ii] / ( dof_2[ii] + dof_1[ii] * f );    
-    result[ii] = incbet( dof_2[ii] / 2.0, dof_1[ii] / 2.0, x);
+    const double delta_dof = dof_1[ii] - dof_2[ii];
+    const double delta_chi = chisq_1[ii] - chisq_2[ii];
+    const double f = delta_chi / delta_dof / (chisq_2[ii] / dof_2[ii]);
+    result[ii] = incbet( dof_2[ii] * 0.5 , delta_dof * 0.5,
+                         (dof_2[ii] / ( dof_2[ii] + delta_dof * f ) ) );
```

Comment:
Do we need to handle possible division by zero?

Reply:
`f` is based on the user input and I guess the user will not input 0 there, but who knows....

Reply:
We have had a case where the _dof < 0_ (see PR #564). So delta_dof, chisq_2 and dof_2 could also be 0 and then should one check for the ranges to function incbet (see: [http://cxc.harvard.edu/sherpa/ahelp/incbet.html](url)).  The fun never stops

Reply:
So we need to add a check for 0.

Reply:
Once upon a time, I had to make enhancements to some CIAO code that was full of global variables.  Unfortunately for me, I modified the code correctly, ie using **local** variables instead of the hideous **global** variables that were written throughout the code.  I was told by @wmclaugh that I have to follow convention of the code and he wants it done **incorrectly**, I kid you not!  Take for example, the following function 
```
def sao_arange(start, stop, step=None):
    """Create a range of values between start and stop.

    See also `numpy.arange` and `numpy.linspace`.

    Parameters
    ----------
    start, stop : float
       The start and stop points.
    step : float or None, optional
       If not given the step size defaults to 1.0.

    Returns
    -------
    vals : NumPy array
       The values start, start + step, ... The last point
       is the first position where start + n * step >= stop,
       which means that it can include a point > stop.

    Examples
    --------

    >>> sao_arange(1, 3)
    array([ 1.,  2.,  3.])

    >>> sao_arange(1, 3, 0.6)
    array([ 1. ,  1.6,  2.2,  2.8,  3.4])

    """

    if step is None:
        return _utils.sao_arange(start, stop)
    else:
        return _utils.sao_arange(start, stop, step)
```
defined in the same file (_sherpa/utils/__init__.py_). Logic tells me that one should check for ```start < stop``` but it is obviously not done here.  So the question now is:  "has the follow convention policy been changed?"